### PR TITLE
Support getting stream_id and track_id of Media

### PR DESCRIPTION
**Pull request by uwuPyxl**
Get the stream_id and track_id from the incoming MediaLine and expose them to users on the resulting Media so they can tell incoming Media apart. Should the incoming Media not have a Msid attribute this falls back to the previous behavior of generating random ids

### CODE REVIEW

**Comment by algesten** [comment]
Hey! Thanks for this!

I was under the impression that msid was deprecated, but I think I'm mistaken. The `a=ssrc` lines are not in fashion anymore, which means getting msid there is not a good idea. But this PR is working with `a=msid`, which seems to still be good API. Is that your understanding too?

**Comment by uwuPyxl** [comment]
I just used the msid attribute since it was most obvious and convenient but yeah it does seem to be intended for this purpose.

File: src/media/mod.rs [hunk 1]
```diff
@@ -206,6 +206,14 @@ impl Media {
     pub(crate) fn msid(&self) -> &Msid {
         &self.msid
     }
+    /// Identifier for the group this Media belongs to.
+    pub fn stream_id(&self) -> &str {
+        &self.msid().stream_id
+    }
+    /// Identifier for this Media. Should be unique for the given stream id.
```
Comment (algesten) [inline src/media/mod.rs:213]: Line break here

File: src/media/mod.rs [hunk 2]
```diff
@@ -206,6 +206,14 @@ impl Media {
     pub(crate) fn msid(&self) -> &Msid {
         &self.msid
     }
+    /// Identifier for the group this Media belongs to.
+    pub fn stream_id(&self) -> &str {
+        &self.msid().stream_id
+    }
```
Comment (algesten) [inline src/media/mod.rs:212]: New line here.

File: src/sdp/data.rs [hunk 3]
```diff
@@ -242,6 +243,15 @@ impl MediaLine {
             // a mid line. This is checked by `check_consistent`.
             .expect("missing a=mid")
     }
+    pub fn msid(&self) -> Option<Msid> {
```
Comment (algesten) [inline src/sdp/data.rs:246]: New line here.

File: src/sdp/data.rs [hunk 4]
```diff
@@ -1116,6 +1126,14 @@ pub struct Msid {
     pub stream_id: String,
     pub track_id: String,
 }
+impl Msid {
```
Comment (algesten) [inline src/sdp/data.rs:1129]: New line here.

**Review by algesten — CHANGES_REQUESTED** [review]
Ok. Let's merge, but 3 small formatting things first.

**Comment by uwuPyxl** [comment]
I've updated the formatting manually, but it should be possible to set up a rustfmt.toml to configure the formatting rules

**Review by algesten — APPROVED** [review]

**Pull request merged**